lib/python/qmk/flashers.py

```diff
@@ -196,6 +196,10 @@ def _flash_uf2(file):
     cli.run(['util/uf2conv.py', '--deploy', file], capture_output=False)
 
 
+def _flash_ubaboot(file):
+    cli.run(['util/ubaboot.py', '--wait', 'write', file], capture_output=False)
```

Comment:
Probably dont need the wait here, as the cli flashing process would have already searched for the vid:pid
```suggestion
    cli.run(['util/ubaboot.py', 'write', file], capture_output=False)
```

Reply:
But that only happens when you use `qmk flash FILE.HEX`; if you run `qmk flash -kb XXXXX -km YYYYY`, it doesn't check if the device is ready.

Reply:
(or maybe it would be better to add a patch for that case too...?)

Reply:
> it doesn't check if the device is ready.

That goes through a different pathway to `flash.mk`, so you are misunderstood how this works.

Reply:
What I mean is that maybe it would be a good idea to add the same generic device check in that other pathway too, and completely remove the waiting code from ubaboot.py, thus leaving the ubaboot.py code as it's currently in upstream. It also would benefit all other programmers.

Reply:
Ok; if the idea is to move everything to CLI, then it makes no sense in adding the check now to the flash.mk pathway.